src/target/source/codegen_c.cc

```diff
@@ -631,8 +632,10 @@ void CodeGenC::PrintVecBinaryOp(const std::string& op, DataType t, PrimExpr lhs,
   }
 }
 
+NameSupply global_name_supply = NameSupply("");
 void CodeGenC::VisitStmt_(const AllocateConstNode* op) {
-  std::string symbol_name = op->buffer_var->name_hint;
+  std::string symbol_name = global_name_supply->FreshName(op->buffer_var->name_hint);
```

Comment:
why was this needed? i think need to initialize the NameSupply from the IRModule

Reply:
Unfortunately, just calling `name_supply_->FreshName(op->buffer_var->name_hint);` does not work - we need to have a global name generator.

Reply:
@gigiblender could you advise what you think is best to do here?

Reply:
i discussed with @gigiblender and in doing so we realized that there is already `var_idmap_` which uses another NameSupply. Does `AllocVarID(op->buffer_var.get())` work here? this parallels what was done in [AllocateNode](https://github.com/apache/tvm/blob/bf86d9f8c25e105a12c1b96ab0a4cc977ee088b7/src/target/source/codegen_c.cc#L857).